src/pages/index.js

```diff
@@ -5,123 +5,260 @@ import PopupWithForm from "../components/PopupWithForm.js";
+  document.querySelector("#profile-description-input").value =
+    newUserInfo.description;
+});
+
+// Functions
+// Functions
+
+function createCard(cardData) {
+  const card = new Card(
+    cardData,
+    "#card-template",
+    handleImageClick,
+    handleDeleteModal,
+    handleLikeCard
+  );
+  const cardElement = card.getView();
+  return cardElement;
 }
 
-function handleImageClick(cardData) {
-  newImagePopup.open(cardData);
+function handleDeleteModal(card) {
+  cardDeletePopup.submitHandler(() => handleDeleteCard(card));
+  cardDeletePopup.open();
 }
 
-function handleAddCardFormSubmit(userInfo) {
-  const name = userInfo.title;
-  const link = userInfo.url;
-  renderCard({ name, link }, cardListElement);
-  addCardFormElement.reset();
-  newCardPopup.close();
-  addFormValidator.disableButton();
+function handleDeleteCard(card) {
+  const cardId = card.gatherCardId();
+  api
+    .cardDelete(cardId)
+    .then(() => {
+      card.handleDelete();
+      //cardDeletePopup.close();
+    })
+    .catch((err) => {
+      console.log(err);
+    });
 }
 
-//Event Listeners
-//Event Listeners
+function handleLikeCard(card) {
+  if (card.isLiked) {
+    api
+      .dislikeCard(card.gatherCardId())
+      .then((res) => {
+        card.handleLike(res.isLiked);
+      })
+      .catch((err) => {
+        console.error(err);
+      });
+  } else {
+    api
+      .likeCard(card.gatherCardId())
+      .then((res) => {
+        card.handleLike(res.isLiked);
+      })
+      .catch((err) => {
+        console.error(err);
+      });
+  }
+}
 
-profileEditBtn.addEventListener("click", prefillProfileData);
+// Form Submission
+// Form Submission
 
-addNewCardButton.addEventListener("click", () => newCardPopup.open());
+function handleAddCardFormSubmit(data) {
+  newCardPopup.renderloading(true);
+  api
+    .newCardAdd({ name: data.title, link: data.url })
+    .then((data) => {
+      const addCardElement = createCard(data);
+      section.addItem(addCardElement);
+      newCardPopup.resetForm();
+      newCardPopup.close();
+      addFormValidator.disableButton();
+    })
+    .catch((err) => {
+      console.log(err);
+    })
+    .finally(() => {
+      newCardPopup.renderloading(false);
+    });
+}
 
-//Event Handlers
-//Event Handlers
+function handleProfileEditSubmit(userData) {
+  editProfilePopup.renderloading(true);
+  api
+    .editProfileInfo(userData.name, userData.description)
+    .then((res) => {
+      userInfo.setUserInfo(res.name, res.about);
+      editFormValidator.disableButton();
+    })
+    .catch((err) => {
+      console.log(err);
+    })
+    .finally(() => {
+      editProfilePopup.renderloading(false);
+      editProfilePopup.close();
+    });
+}
+
+function handleAvatarFormSubmit(userData) {
+  avatarPopup.renderloading(true);
+  api
+    .avatarImageUpdate(userData)
+    .then((res) => {
+      userInfo.setAvatar(res.avatar);
+    })
+    .catch((err) => {
+      console.log(err);
```

Comment:
Good job that you `catch` possible errors at the end of server requests. You can simplify it with `.catch(console.error);` because `console.error` is meant to log errors